### ISSUE TITLE
Playlist will get longer with autoload.lua and playlist-shuffle command in some cases

### ISSUE THREAD

**Issue by snylonue** [mpv-player/mpv#7552]
### Important Information

Provide following Information:
- mpv version
```
mpv 0.32.0-258-g281f5c63c1 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on Sun Mar 15 22:10:08 +08 2020
FFmpeg library versions:
   libavutil       56.42.100
   libavcodec      58.75.100
   libavformat     58.41.100
   libswscale      5.6.101
   libavfilter     7.77.100
   libswresample   3.6.100
FFmpeg version: git-2020-03-14-425b6a9a2
```
- Windows Version `10.0.18363.720`
- Source of the mpv binary https://sourceforge.net/projects/mpv-player-windows

- Possible screenshot or video of visual glitches

The initial playlist
![Snipaste_2020-03-22_19-45-21](https://user-images.githubusercontent.com/48825598/77248685-376c3300-6c76-11ea-9f77-d2c28775d2c1.png)
After some operations below
![Snipaste_2020-03-22_19-45-39](https://user-images.githubusercontent.com/48825598/77248687-3a672380-6c76-11ea-8152-51f92704b04a.png)

### Reproduction steps

Install script autoload.
Install the script like below(may not work)
```javascript
mp.register_event('start-file', function() { mp.command('playlist-shuffle'); });
```
Open a playlist with multiple files.

### Expected behavior

The playlist is shuffled and remains

### Actual behavior

The playlist gets longer.

### Log file

[mpv.log](https://github.com/mpv-player/mpv/files/4397124/mpv.log)

**Comment by Samillion** [comment]
Can confirm. I have noticed that this happens (mostly) if the files have non-English characters and some symbols. No clue if it's related in any way.

With a full English characters list, it never happens.

**Comment by snylonue** [comment]
May this problem cause by filename encoding?
It seems to be solved since `0.32.0-383-ga09c7691d7`

**Comment by KAELINS** [comment]
This is still happening... the playlist keeps getting longer and longer
